packages/api/src/external/carequality/document/process-outbound-document-query-resps.ts

```diff
@@ -147,44 +130,15 @@ export async function processOutboundDocumentQueryResps({
     // Since we have most of the document contents when doing the document query,
     // we will store this in FHIR and then upsert the reference to the s3 object in FHIR
     // when doing the doc retrieval
+    const docRefsWithMetriportId = filterDocRefsWithMetriportId(docsToDownload);
```

Comment:
I understand we're doing this just because the type on `storeInitDocRefInFHIR` requires docRefs w/ Metriport ID, and that all items on `docsToDownload` should have the Metriport ID, right?

I'd try to fix that by specializing the type returned by `getRespWithDocsToDownload`:
- It's not good that we have to execute code in runtime that's only needed to support the compilation time (types);
- Reading the code one thinks we're actually filtering items out, and that potentially could lead to docs downloaded that are not stored on FHIR.

Something like:

```ts
type DqRespWithDocRefsWithMetriportId = OutboundDocumentQueryResp & {
  documentReference: DocumentReferenceWithMetriportId[];
};
```

Reply:
Yea thats a valid point - updated